src/parser/jobs/drk/GCDs.js

```diff
@@ -0,0 +1,138 @@
+import React, {Fragment} from 'react'
+import {Accordion, Message} from 'semantic-ui-react'
+import Rotation from 'components/ui/Rotation'
+import {ActionLink} from 'components/ui/DbLink'
+import {Suggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+import Module from 'parser/core/Module'
+import ACTIONS from 'data/ACTIONS'
+//import STATUSES from 'data/STATUSES'
+
+const GCD_COMBO_DURATION = 10000
+
+// GCD Combo affecting actions.
+const GCD_COMBO_ACTIONS = [
+	ACTIONS.HARD_SLASH.id,
+	ACTIONS.SYPHON_STRIKE.id,
+	ACTIONS.SOULEATER.id,
+	ACTIONS.SPINNING_SLASH.id,
+	ACTIONS.POWER_SLASH.id,
+	ACTIONS.UNMEND.id,
+	ACTIONS.ABYSSAL_DRAIN.id,
+]
+
+// GCD Combo Chain.  If this chain isn't respected, all additional effects of attacks are discarded.
+const GCD_COMBO_CHAIN = {
+	[ACTIONS.HARD_SLASH.id]: {
+		next: [ACTIONS.SYPHON_STRIKE.id, ACTIONS.SPINNING_SLASH.id]},
+	[ACTIONS.SYPHON_STRIKE.id]: {
+		next: [ACTIONS.SOULEATER.id]},
+	[ACTIONS.SOULEATER.id]: {
+		next: undefined},
+	[ACTIONS.SPINNING_SLASH.id]: {
+		next: [ACTIONS.POWER_SLASH.id]},
+	[ACTIONS.POWER_SLASH.id]: {
+		next: undefined},
+}
+
+export default class GCDs extends Module {
+	static handle = 'gcds'
+	static title = 'GCD Combo'
+	static dependencies = [
+		'cooldowns',
+		'suggestions',
+	]
+
+	// gcd combo
+	_GCDComboActive = false
+	_lastComboAction = undefined
+	_last3eventsAndCurrent = []
+	_lastComboGCDTimeStamp = undefined
+	_GCDChainDrops = []
+
+	inGCDCombo() {
+		return this._GCDComboActive
+	}
+
+	constructor(...args) {
+		super(...args)
+		this.addHook('cast', {by: 'player'}, this._onCast)
+	}
+
+	_onCast(event) {
+		const abilityId = event.ability.guid
+		// check combo status
+		if (GCD_COMBO_ACTIONS.includes(abilityId)) {
+			this._last3eventsAndCurrent.push(event)
+			if (
+				(this._lastComboAction !== undefined && this._lastComboGCDTimeStamp !== undefined) &&
+				(event.timestamp - this._lastComboGCDTimeStamp < GCD_COMBO_DURATION) &&
+				(GCD_COMBO_CHAIN.hasOwnProperty(this._lastComboAction))
+			) {
+				const entry = GCD_COMBO_CHAIN[this._lastComboAction]
```

Comment:
This seems overly complicated because of the fact you forward referenced.

Instead of
Looking up the last GCD action -> grabbing it from the mapping -> checking if it has a next combo array -> checking if current action is in it.

You could just
Grabbing current GCD Action from mapping -> check if last action is `combosFrom`

Reply:
the two arrays served different purposes.  Merged together in object property notation to make things more clear and remove the double iteration